File: ios-rider-app/MePontoRider/Models/RiderTier.swift
import SwiftUI

// MePonto rider membership tier — ported 1:1 from app/rider-app/page.tsx
// (getRiderTierScore + getRiderTierByScore). The score blends today's
// performance with the rider's consistency/incident history; thresholds
// map to 1★ → 2★ → 3★ → Gold → Diamond.

struct RiderTierInfo {
    let key: String      // base | green | orange | gold | diamond
    let label: String    // 1 estrela / 2 estrelas / 3 estrelas / Gold / Diamond
    let stars: Int
    let benefit: String
    let nextTarget: String

    // Tier gradient + accent for the membership card.
    var gradient: [Color] {
        switch key {
        case "diamond": return [Theme.color(0x07111f), Theme.color(0x123b53), Theme.color(0xbef0f7)]
        case "gold":    return [Theme.color(0x1d1202), Theme.color(0x9a5b08), Theme.color(0xffb238)]
        case "orange":  return [Theme.color(0x120b05), Theme.color(0x783900), Theme.color(0xff7a00)]
        case "green":   return [Theme.color(0x06150e), Theme.color(0x0f5130), Theme.color(0x20a65a)]
        default:        return [Theme.color(0x050505), Theme.color(0x161616)]
        }
    }

    var accent: Color {
        switch key {
        case "diamond": return Theme.color(0xa8f3ff)
        case "gold":    return Theme.color(0xffe2a3)
        case "orange":  return Theme.color(0xffb16a)
        case "green":   return Theme.color(0x91e8b4)
        default:        return Theme.color(0xff7a00)
        }
    }
}

enum RiderTier {

    // Today's performance inputs (mirrors `performanceToday` in the web app).
    static func score(ar: Int, nightShiftCount: Int, incidentCount: Int,
                      orders: Int = 18, tshHours: Double = 7.4,
                      acceptanceToday: Int = 96, caaOrders: Int = 5) -> Int {
        let orderScore = Double(min(orders, 24)) * 1.2
        let tshScore = min(tshHours, 10) * 2.2
        let arScore = Double(max(0, acceptanceToday - 70)) * 1.4
        let caaScore = Double(min(caaOrders, 8)) * 3
        let consistencyScore = Double(min(nightShiftCount, 18)) * 0.8
        let incidentPenalty = Double(incidentCount) * 8
        return Int((orderScore + tshScore + arScore + caaScore + consistencyScore - incidentPenalty + 12).rounded())
    }

    static func byScore(_ score: Int) -> RiderTierInfo {
        if score >= 108 {
            return RiderTierInfo(key: "diamond", label: "Diamond", stars: 5,
                                 benefit: "Max perks", nextTarget: "Topo Diamond")
        }
        if score >= 100 {
            return RiderTierInfo(key: "gold", label: "Gold", stars: 4,
                                 benefit: "Fila premium", nextTarget: "\(108 - score) pts → Diamond")
        }
        if score >= 86 {
            return RiderTierInfo(key: "orange", label: "3 ★", stars: 3,
                                 benefit: "Bônus pontos", nextTarget: "\(100 - score) pts → Gold")
        }
        if score >= 72 {
            return RiderTierInfo(key: "green", label: "2 ★", stars: 2,
                                 benefit: "Mais missões", nextTarget: "\(86 - score) pts → 3 ★")
        }
        return RiderTierInfo(key: "base", label: "1 ★", stars: 1,
                             benefit: "Base ativa", nextTarget: "\(max(0, 72 - score)) pts → 2 ★")
    }
}

// Rider membership / identity profile (会员资料), mirrors fields on the
// web Rider model: ponto (网点), leader (队长), bairro (片区), 99 ID.
struct MembershipProfile {
    var name: String
    var ponto: String
    var leader: String
    var bairro: String
    var ninetyNineId: String
    var ar: Int
    var nightShiftCount: Int
    var incidentCount: Int
    // Identity / payout details the rider can view and complete in the app.
    var cpf: String = ""
    var phone: String = ""
    var pix: String = ""

    var tierScore: Int { RiderTier.score(ar: ar, nightShiftCount: nightShiftCount, incidentCount: incidentCount) }
    var tier: RiderTierInfo { RiderTier.byScore(tierScore) }

    /// PIX + CPF + phone are required to receive payouts.
    var isComplete: Bool {
        !cpf.trimmingCharacters(in: .whitespaces).isEmpty &&
        !pix.trimmingCharacters(in: .whitespaces).isEmpty &&
        !phone.trimmingCharacters(in: .whitespaces).isEmpty
    }
}

File: android-rider-app/app/src/main/java/com/meponto/rider/data/RiderTier.kt
package com.meponto.rider.data

import androidx.compose.ui.graphics.Color
import kotlin.math.roundToInt

/**
 * MePonto rider membership tier — ported 1:1 from app/rider-app/page.tsx
 * (getRiderTierScore + getRiderTierByScore). The score blends today's
 * performance with the rider's consistency/incident history; thresholds map to
 * 1★ → 2★ → 3★ → Gold → Diamond.
 */
data class RiderTierInfo(
    val key: String,     // base | green | orange | gold | diamond
    val label: String,   // 1 ★ / 2 ★ / 3 ★ / Gold / Diamond
    val stars: Int,
    val benefit: String,
    val nextTarget: String,
) {
    // Tropical/Noite v2 tier language: diamond = electric blue night sky,
    // gold = Rio sunset (laranja→dourado), orange = hot sunset pink,
    // green = tropical jungle, base = deep noite violet.
    val gradient: List<Color>
        get() = when (key) {
            "diamond" -> listOf(Color(0xFF0E1B4D), Color(0xFF2D6BFF), Color(0xFF8AE9FF))
            "gold" -> listOf(Color(0xFF9A3D00), Color(0xFFFF6A3D), Color(0xFFFFC400))
            "orange" -> listOf(Color(0xFF5C1030), Color(0xFFFF4D8D), Color(0xFFFF8A3D))
            "green" -> listOf(Color(0xFF06150E), Color(0xFF0B5C3B), Color(0xFF00A868))
            else -> listOf(Color(0xFF12081F), Color(0xFF2E2044))
        }

    val accent: Color
        get() = when (key) {
            "diamond" -> Color(0xFFA8F3FF)
            "gold" -> Color(0xFFFFE18A)
            "orange" -> Color(0xFFFFD1E1)
            "green" -> Color(0xFF9FD9BE)
            else -> Color(0xFFFFC400)
        }
}

object RiderTier {

    // Today's performance inputs (mirrors `performanceToday` in the web app).
    fun score(
        ar: Int,
        nightShiftCount: Int,
        incidentCount: Int,
        orders: Int = 18,
        tshHours: Double = 7.4,
        acceptanceToday: Int = 96,
        caaOrders: Int = 5,
    ): Int {
        val orderScore = minOf(orders, 24) * 1.2
        val tshScore = minOf(tshHours, 10.0) * 2.2
        val arScore = maxOf(0, acceptanceToday - 70) * 1.4
        val caaScore = minOf(caaOrders, 8) * 3.0
        val consistencyScore = minOf(nightShiftCount, 18) * 0.8
        val incidentPenalty = incidentCount * 8.0
        return (orderScore + tshScore + arScore + caaScore + consistencyScore - incidentPenalty + 12).roundToInt()
    }

    fun byScore(score: Int): RiderTierInfo = when {
        score >= 108 -> RiderTierInfo("diamond", "Diamond", 5, "Max perks", "Topo Diamond")
        score >= 100 -> RiderTierInfo("gold", "Gold", 4, "Fila premium", "${108 - score} pts → Diamond")
        score >= 86 -> RiderTierInfo("orange", "3 ★", 3, "Bônus pontos", "${100 - score} pts → Gold")
        score >= 72 -> RiderTierInfo("green", "2 ★", 2, "Mais missões", "${86 - score} pts → 3 ★")
        else -> RiderTierInfo("base", "1 ★", 1, "Base ativa", "${maxOf(0, 72 - score)} pts → 2 ★")
    }
}

/**
 * Rider membership / identity profile (会员资料), mirrors fields on the web
 * Rider model: ponto (网点), leader (队长), bairro (片区), 99 ID.
 */
data class MembershipProfile(
    val name: String,
    val ponto: String,
    val leader: String,
    val bairro: String,
    val ninetyNineId: String,
    val ar: Int,
    val nightShiftCount: Int,
    val incidentCount: Int,
    // Identity / payout details (editable; required before requesting a payout).
    val cpf: String = "",
    val phone: String = "",
    val pix: String = "",
    val birthday: String = "",
) {
    val tierScore: Int get() = RiderTier.score(ar, nightShiftCount, incidentCount)
    val tier: RiderTierInfo get() = RiderTier.byScore(tierScore)
    /** Payout requires CPF + PIX + phone all present. */
    val isComplete: Boolean get() = cpf.isNotBlank() && pix.isNotBlank() && phone.isNotBlank()
}
